src/modules/PasswordRecovery/PasswordRecovery.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import './PasswordRecovery.scss';
+import { ChangeInputEvent, SubmitFormEvent } from '../../@types/types';
+import CommonButton from '../../UI/CommonButton/CommonButton';
+import {
+  validMail,
+  validOnlyNumber,
+  validPassword,
+  validRecoveryCode,
+} from '../../helpers';
+import MainApiService from '../../api/MainApiService';
+
+export default function PasswordRecovery() {
+  const [recoveryMail, setRecoveryMail] = useState('');
+  const [recoveryCode, setRecoveryCode] = useState('');
+  const [recoveryNewPassword, setRecoveryNewPassword] = useState('');
+  const [recoveryNewPassword2, setRecoveryNewPassword2] = useState('');
+
+  const [mailFormVisible, setMailFormVisible] = useState(true);
+  const [recoveryCodeFormVisible, setRecoveryCodeFormVisible] = useState(false);
+  const [newPasswordFormVisible, setNewPasswordFromVisible] = useState(false);
+
+  const navigate = useNavigate();
+
+  const onMailChage = (event: ChangeInputEvent) => {
+    event.preventDefault();
+    const target = event.target as HTMLInputElement;
+    setRecoveryMail(target.value);
+  };
+
+  const onRecoveryCodeChage = (event: ChangeInputEvent) => {
+    const newValue = event.target.value;
+    if (validOnlyNumber(newValue)) setRecoveryCode(event.target.value);
+  };
+
+  const onNewPasswordChage = (event: ChangeInputEvent) => {
+    setRecoveryNewPassword(event.target.value);
+  };
+
+  const onNewPassword2Chage = (event: ChangeInputEvent) => {
+    setRecoveryNewPassword2(event.target.value);
+  };
+
+  const submitMail = async (event: SubmitFormEvent) => {
+    event.preventDefault();
+    if (validMail(recoveryMail)) {
+      const json = await MainApiService.passwordRecoveryMail({
+        email: recoveryMail,
+      });
+      if (json.statusCode !== 0) {
+        setMailFormVisible(false);
+        setRecoveryCodeFormVisible(true);
+      } else {
+        console.log('Mail is not find');
+      }
+    } else console.log('Invalid mail');
```

Comment:
Use console only for tests. If we don't have any right way to show errors for users - create it or add TODO comment and make the task:
```
// TODO: create readable error component for users
```